server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/DomainsRoutesTest.java

```diff
@@ -157,10 +160,20 @@ public void getDomainsShouldDisplayAddedDomains() {
 
         @Test
         public void putShouldReturnUserErrorWhenNameContainsAT() {
-            when()
+            Map<String, Object> errors = when()
                 .put(DOMAIN + "@" + DOMAIN)
             .then()
-                .statusCode(400);
+                .statusCode(400)
+                .contentType(ContentType.JSON)
+                .extract()
+                .body()
+                .jsonPath()
+                .getMap(".");
+
+            assertThat(errors)
+                .containsEntry("statusCode", 400)
+                .containsEntry("type", "InvalidArgument")
+                .containsEntry("message", "Invalid request for domain creation domain@domain");
```

Comment:
It is sad that contrary to the name of this test "putShouldReturnUserErrorWhenNameContainsAT" the error message does not let someone easy understand that the request fail because of a ``@`` in the username

Reply:
I would like to keep all the function and test function as they are. I just add more assertion error information in case of error.

Reply:
I get you point of view but if it is easy to fix, it is the good opportunity to improve James.
Clear error message are really important I think for the UX